src/wallet/scriptpubkeyman.cpp

```diff
@@ -86,19 +86,48 @@ void LegacyScriptPubKeyMan::RewriteDB()
 {
 }
 
+static int64_t GetOldestKeyTimeInPool(const std::set<int64_t>& setKeyPool, WalletBatch& batch) {
+    if (setKeyPool.empty()) {
+        return GetTime();
+    }
+
+    CKeyPool keypool;
+    int64_t nIndex = *(setKeyPool.begin());
+    if (!batch.ReadPool(nIndex, keypool)) {
+        throw std::runtime_error(std::string(__func__) + ": read oldest key in keypool failed");
+    }
+    assert(keypool.vchPubKey.IsValid());
+    return keypool.nTime;
+}
+
 int64_t LegacyScriptPubKeyMan::GetOldestKeyPoolTime()
 {
-    return GetTime();
+    LOCK(cs_KeyStore);
+
+    WalletBatch batch(*m_database);
+
+    // load oldest key from keypool, get time and return
+    int64_t oldestKey = GetOldestKeyTimeInPool(setExternalKeyPool, batch);
+    if (IsHDEnabled() && CanSupportFeature(FEATURE_HD_SPLIT)) {
+        oldestKey = std::max(GetOldestKeyTimeInPool(setInternalKeyPool, batch), oldestKey);
+        if (!set_pre_split_keypool.empty()) {
+            oldestKey = std::max(GetOldestKeyTimeInPool(set_pre_split_keypool, batch), oldestKey);
+        }
+    }
+
+    return oldestKey;
 }
 
 size_t LegacyScriptPubKeyMan::KeypoolCountExternalKeys()
 {
-    return 0;
+    LOCK(cs_KeyStore);
+    return setExternalKeyPool.size() + set_pre_split_keypool.size();
 }
 
 unsigned int LegacyScriptPubKeyMan::GetKeypoolSize() const
 {
-    return 0;
+    LOCK(cs_KeyStore);
+    return setInternalKeyPool.size() + setExternalKeyPool.size() + set_pre_split_keypool.size();
```

Comment:
In commit "Implement GetOldestKeyPoolTime, KeypoolCountExternalKeys, and GetKeypoolSize in LegacyScriptPubKeyMan" (26c1ebe6a7bea812860636f9d7316fc71b8d90dc)

Is it a bugfix to include set_pre_split_keypool size in this return value? Wallet method doesn't do this, so behavior seems to change in later commit 937cf1e2265ace44147475dc9da8a5ec0b1dda47

https://github.com/bitcoin/bitcoin/blob/f4a0d27e85754d60804ffa36e415b67c263180b9/src/wallet/wallet.h#L1251-L1255

Reply:
Yes, that is a bug fix.